### PR TITLE
feat: redesign of netlify dev, netlify init, and some kind of style guide

### PR DESCRIPTION
**Pull request by whitep4nth3r**
# Please do not look at this yet. 

This is a draft PR to check the tests in CI because running the tests locally is not very nice.

🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes #<replace_with_issue_number>

Related issues: 
https://github.com/netlify/cli/issues/164
https://github.com/netlify/pillar-workflow/issues/201

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: src/lib/edge-functions/registry.cjs [hunk 1]
```diff
@@ -396,12 +406,15 @@ class EdgeFunctionsRegistry {
...
 
   async scan(directories) {
+    logH2({ message: 'Loading netlify edge functions' })
     const functions = await this.bundler.find(directories)
 
     functions.forEach((func) => {
       EdgeFunctionsRegistry.logAddedFunction(func)
     })
 
+    functions.push(log())
```
Comment (danez) [inline src/lib/edge-functions/registry.cjs:416]: This seems wrong, pushing a log line or undefined (?) to the functions array.

File: src/lib/spinner.cjs [hunk 2]
```diff
@@ -1,17 +1,94 @@
...
+  if (noColors) {
+    const colorlessChalk = new ChalkInstance({ level: 0 })
+    return colorlessChalk
+  }
+  return new ChalkInstance()
+}
+/** The parsed process argv without the binary only arguments and flags */
+const argv = process.argv.slice(2)
+
+const chalk = safeChalk(argv.includes('--json'))
```
Comment (danez) [inline src/lib/spinner.cjs:29]: We could simply use `chalk` from command-helpers here I think?

File: src/lib/edge-functions/registry.cjs (hunk 1 above)
Comment (whitep4nth3r) [reply src/lib/edge-functions/registry.cjs:416]: I specifically wrote at the top of the PR for no one to look at this yet :)

This is the most drafty PR I have ever opened and I haven't even got my bearings on the repo yet. It's all a draft and I will be addressing things like this.

Thanks for your comment, but please no more until the PR is out of draft.

Comment (danez) [reply src/lib/edge-functions/registry.cjs:416]: I know I was just looking at what could be the reason for the timeouts, and only found this. 
*disappears in smoke*

File: src/lib/spinner.cjs (hunk 2 above)
Comment (whitep4nth3r) [reply src/lib/spinner.cjs:29]: I had some kind of warning in my terminal about not using safeChalk at this point, hence the copy paste. I'll look into it again.

File: src/lib/edge-functions/registry.cjs (hunk 1 above)
Comment (whitep4nth3r) [reply src/lib/edge-functions/registry.cjs:416]: Oh interesting! Oh yeah this a horrendous line of code isn't it :D Thanks for spotting it!

**Pull request closed without merging**